pom.xml

```diff
@@ -64,15 +64,15 @@
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-        <jenkins.version>2.289.1</jenkins.version>
+        <jenkins.version>2.387.3</jenkins.version>
```

Comment:
This bump causes `MockCloudTest.inbound` to fail: agents keep on being provisioned, but never used. (Regardless of the `jenkins.version` in `mock-slave-plugin/pom.xml`! Discovered by bisecting `bom` to https://github.com/jenkinsci/bom/pull/2838 and so on.) This is true as of 2.357.

```
INFO: Could not locate server among [http://localhost:43027/jenkins/]; waiting 10 seconds before retry
java.io.IOException: http://localhost:43027/jenkins/tcpSlaveAgentListener/ appears to be publishing an invalid X-Instance-Identity.
	at org.jenkinsci.remoting.engine.JnlpAgentEndpointResolver.resolve(JnlpAgentEndpointResolver.java:277)
	at hudson.remoting.Engine.innerRun(Engine.java:809)
	at hudson.remoting.Engine.run(Engine.java:563)
```

suggests https://github.com/jenkinsci/jenkins/pull/6585 is involved; perhaps something weird with Maven dependency handling. Adding `instance-identity` as a test dep solves the issue.